src/main/java/emissary/util/search/ByteMatcher.java

```diff
@@ -252,6 +335,32 @@ public int indexIgnoreCase(byte[] pattern, int beginIndex, int endIndex) {
 
     }
 
+    /**
+     * This method finds a pattern in the text from {@code beginIndex} to {@code endIndex} and returns the offset list
+     * ignoring upper/lower case
+     *
+     * @param pattern bytes to find
+     * @param beginIndex start index
+     * @param endIndex the index to stop searching at, exclusive
+     *
+     * @return list of positions
+     */
+    public ArrayList<Integer> indexListIgnoreCase(byte[] pattern, int beginIndex, int endIndex) {
+
+        // Impossible to find under these conditions
+        if (mydata == null || beginIndex > (mydata.length - pattern.length) || endIndex > mydata.length) {
+            return EMPTYLIST;
```

Comment:
Update other occurrences as well.
```suggestion
    public List<Integer> indexListIgnoreCase(byte[] pattern, int beginIndex, int endIndex) {

        // Impossible to find under these conditions
        if (mydata == null || beginIndex > (mydata.length - pattern.length) || endIndex > mydata.length) {
            return Collections.emptyList();
```